node/tests/utils.rs

```diff
@@ -525,47 +523,56 @@ impl MASQNode {
     ) -> Vec<String> {
         fn retain_unique_default_args(
             default_args: Vec<String>,
-            config_args: &BTreeMap<String, Option<String>>,
-        ) -> BTreeMap<String, Option<String>> {
-            MASQNode::virtual_arg_pairs(default_args)
+            config_args: &HashMap<String, PositionedArg>,
+        ) -> HashMap<String, PositionedArg> {
+            MASQNode::virtual_arg_pairs_with_remembered_position(default_args)
                 .into_iter()
                 .flat_map(|(arg_name, value)| {
-                    let res = config_args
+                    config_args
                         .keys()
                         .contains(&&arg_name)
                         .not()
-                        .then_some((arg_name, value));
-                    res
+                        .then_some((arg_name, value))
                 })
                 .collect()
         }
-        fn test_uniqueness_for_config_args(config_args_as_tuples: &[(String, Option<String>)]) {
+        fn test_uniqueness_for_config_args(config_args_as_tuples: &[(String, PositionedArg)]) {
             let len_before = config_args_as_tuples.len();
             let config_args_to_test: Vec<_> = config_args_as_tuples
```

Comment:
This might look a little simpler if you took this Vector of tuples and made a HashMap out of it: that should automatically deduplicate the keys, and you could just compare the lengths of the Vector and the HashMap.